### ISSUE TITLE
0.17.0 is completely broken on Mac

### ISSUE THREAD

**Issue by ethaniel** [feross/webtorrent-desktop#1001]
Can't use the interface, can't resize app, laptop is now overheating and fan is extremely high.

**Comment by feross** [comment]
@ethaniel Whoa, this sounds terrible. Can you go to the `~/Library/Application Support/WebTorrent` folder and find the `config.json` file and email it to me, so I can investigate this?

**Comment by feross** [comment]
Also, can you start the app and try pressing `Command` + `Option` + `I` to bring up the inspector. If it opens, go to the Console tab and copy the text and paste it here.

**Comment by feross** [comment]
I'm seeing lots of this error in our telemetry logs:

```
Processes: window, platforms: darwin linux win32, versions: pre-0.12 0.17.0
TypeError: Cannot read property 'getHostNode' of null
    at Object.getHostNode (.../node_modules/react/lib/ReactReconciler.js:64:28)
    at ReactCompositeComponentWrapper.getHostNode (.../node_modules/react/lib/ReactCompositeComponent.js:383:28)
    at Object.getHostNode (.../node_modules/react/lib/ReactReconciler.js:64:29)
    at Object.updateChildren (.../node_modules/react/lib/ReactChildReconciler.js:114:48)
    at ReactDOMComponent._reconcilerUpdateChildren (.../node_modules/react/lib/ReactMultiChild.js:215:28)
    at ReactDOMComponent._updateChildren (.../node_modules/react/lib/ReactMultiChild.js:314:31)
    at ReactDOMComponent.updateChildren (.../node_modules/react/lib/ReactMultiChild.js:301:12)
    at ReactDOMComponent._updateDOMChildren (.../node_modules/react/lib/ReactDOMComponent.js:942:12)
    at ReactDOMComponent.updateComponent (.../node_modules/react/lib/ReactDOMComponent.js:760:10)
    at ReactDOMComponent.receiveComponent (.../node_modules/react/lib/ReactDOMComponent.
```

This means an exception is getting thrown somehow and React is getting into a bad state. In the past when I've seen this error, the entire UI just disappears. Not good.

We should fix this ASAP and do a release.

**Comment by ethaniel** [comment]
I have huge downloaded files (10 files, like 20gb each) and I've noticed that when I start the app the UI doesn't respond for a minute or so. System slow and fans spinning fast. (I assume that it is rehashing the files). Noticed in 0.10.0 once I downgraded from 0.17.0. Do you still need the config file?

**Comment by ethaniel** [comment]
The app updated itself back to 0.17.0 and now it's back to having problems.

**Comment by ethaniel** [comment]
So let's recap.

0.10.0 is slow for 1-2 minutes and then works fine.
0.17.0 & 0.17,1 are slow nonstop for over 5 minutes (I haven't tested longer).

The app is updating itself, so I have to reinstall 0.10.0 every time I want to watch something.

**Comment by feross** [comment]
The overall app slowness is a known issue that we want to fix before v1.0. But let's fix the unusable v0.17.0 for you first.

Can you confirm a few things about v0.17.1 for me?

- Does this only happen at startup?
- Does the UI show up? Or is it just a blank gray/black screen?
- Can you go to the `~/Library/Application Support/WebTorrent` folder and find the `config.json` file and email it to me, so I can investigate this? You can find my email on my GitHub profile.

**Comment by ethaniel** [comment]
That's weird. Today it upgraded to 0.17.1 again and when I started it, everything was back to normal. I also deleted multiple huge movies yesterday, so that could be the case. 

Closed for now.
